File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        Map<Integer, Integer>map1 = new HashMap<>();
        for(int num : nums)
        {
            if(map1.containsKey(num))
            {
                return true;
            }
            map1.put(num, 1);
        }
        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        #approach 1 : 2 loops, n^2
        #approach 2: store in set or hashmap
        numCount = set()
        for num in nums:
            if num in numCount:
                return True
            numCount.add(num)
        return False